File: Week_04/PPMH_BT8/PPMH_BT8.py
def main ():
    N = int(input("Enter N: "))
    for n in range(1, N + 1):
        r = (n ** 3 - 4 * n) % 3
        print(r)

        if n % 3 == 0:
            print("Divisible by 3")
        elif n % 3 == 1:
            print("Remainder 1 when divided by 3")
        elif n % 3 == 2:
            print("Remainder 2 when divided by 3")

if __name__ == "__main__":
    main()

File: Week_04/PPMH_BT8/PPMH_BT8.cpp
#include <iostream>
using namespace std;

int main(){
    int N;
    cin >> N;
    for(int n = 1; n <= N; ++n){
        int r =  (n * n * n - 4 * n) % 3;
        cout << r << endl;

        if (n % 3 == 0){
            cout << "n is a multiple of 3" << endl;
        } else if (n % 3 == 1){
            cout << "n gives a remainder of 1 when divided by 3" << endl;
        } else if (n % 3 == 2){
            cout << "n gives a remainder of 2 when divided by 3" << endl;

        }
    }
}